docs/book/src/developer/providers/contracts/infra-machinepool.md

```diff
@@ -0,0 +1,525 @@
+ // +optional
+ Ready bool `json:"ready"`
+}
+```
+
+### InfraMachinePool: providerID
+
+Each InfraMachinePool MAY specify a provider ID on `spec.providerID` that can be used to identify the InfraMachinePool.
+
+```go
+type FooMachinePoolSpec struct {
+    // providerID is the identification ID of the FooMachinePool.
+    // +optional
+	// +kubebuilder:validation:MinLength=1
+	// +kubebuilder:validation:MaxLength=512
+	ProviderID string `json:"providerID,omitempty"`
+    
+    // See other rules for more details about mandatory/optional fields in InfraMachinePool spec.
+    // Other fields SHOULD be added based on the needs of your provider.
+}
+```
+
+NOTE: To align with API conventions, we recommend since the v1beta2 contract that the `ProviderID` field should be
+of type `string`.
+
+
+### InfraMachinePool: providerIDList
+
+Each InfraMachinePool MUST supply a list of the identification IDs of the machine instances managed by the machine pool by storing these in `spec.providerIDList`.
+
+```go
+type FooMachinePoolSpec struct {
+    // ProviderIDList is the list of identification IDs of machine instances managed by this Machine Pool
+	// +optional
+	ProviderIDList []string `json:"providerIDList,omitempty"`
+    
+    // See other rules for more details about mandatory/optional fields in InfraMachinePool spec.
+    // Other fields SHOULD be added based on the needs of your provider.
+}
+```
+
+Cluster API uses this list to determine the status of the machine pool and to know when replicas have been deleted, at which point the Node will be deleted.
+
+### InfraMachinePool: ready
+
+Each provider MUST indicate when then the InfraMachinePool has been complteley provisioned by setting `staus.ready` to **true**. This indicates to Cluster API that the infrastructure is ready and that it can continue with its processing. The value retuned here is stored in the MachinePool's `status.infraStructureReady` field.
+
+```go
+type FooMachinePoolStatus struct {
+	// Ready is true when the provider resource is ready.
+	// +optional
+	Ready bool `json:"ready"`
+    
+    // See other rules for more details about mandatory/optional fields in InfraMachinePool status.
+    // Other fields SHOULD be added based on the needs of your provider.
+}
+```
+
+When `ready` becomes true the phase of the MachinePool changes from **provisioning** to **provisioned**. Its also the signal that the providerIDList and replica status fields should be set on the MachinePool.
+
+### InfraMachinePool: pausing
+
+Providers SHOULD implement the pause behaviour for every object with a reconciliation loop. This is done by checking if `spec.paused` is set on the Cluster object and by checking for the `cluster.x-k8s.io/paused` annotation on the InfraMachinePool object.
+
+If implementing the pause behavior, providers SHOULD surface the paused status of an object using the Paused condition: `Status.Conditions[Paused]`.
+
+### InfraMachinePool: conditions
```

Comment:
Even if not directly related to this PR, I strongly reccomand to prioritise work on MP conditions aiming at the target state defined in https://github.com/kubernetes-sigs/cluster-api/blob/main/docs/proposals/20240916-improve-status-in-CAPI-resources.md#machinepool-newconditions

Currently status.conditions are not computed, and status.deprecated.v1beta1.conditions will go away soon.

Also, If I'm not wrong
- Conditions from InfraMachinePool are surfaced in the deprecated conditions, but they only support v1beta1 condition format (and this is not aligned with the content of this paragraph)
- MP is setting the UpToDateCondition on controlled machines always to true, which seems not ideal

Reply:
I agree, i will ensure that this is added to MP working group list as a priority.